src/interpreter/index.ts

```diff
@@ -446,19 +448,25 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         let requiredType = typeDesignators[name.charAt(name.length - 1)];
 
         if (requiredType && requiredType !== value.kind) {
-            return this.addError(
-                new TypeMismatch({
-                    message: `Attempting to assign incorrect value to statically-typed variable '${name}'`,
-                    left: {
-                        type: requiredType,
-                        location: statement.name.location,
-                    },
-                    right: {
-                        type: value,
-                        location: statement.value.location,
-                    },
-                })
-            );
+            if (requiredType === ValueKind.Int64 && value.kind === ValueKind.Int32) {
+                value = new Int64(value.getValue());
```

Comment:
Will this cast all 32 bit integers to 64 bits even small numbers as `1`?

Reply:
No, it shouldn't because we have above if statement:`if (requiredType && requiredType !== value.kind) {`. 
So, if we try to write an `Integer` value into a variable with an `Integer` type then we get `Integer`:
`someInteger = 1` ' will be integer (`requiredType` and `value.kind` are the same)

The problem was that when we tried to write into `LongInteger` variable, value with an `Integer` type, it crashed due to type mismatch:
`someLongInt& = 1 `' crashes while running tests but works on Roku
By these changes, we allow setting `Integer` value into a variable with `LongInteger` type, the same with `float` and `Double`.

Reply:
Oh I see, what you explain makes sense but I wonder if this should be right place to do that check. Maybe @sjbarag can help us here

Reply:
This is probably the only reasonable place to do this kind of conversion for assignments, but I suspect we're missing a few cases (do we need `foo# = 1`?).

We currently do this sort of type promotion in `Callable.ts` (if we expected a Double but received a Float, convert the received Float to a Double) — in fact, the next line basically duplicates the logic from `Callable#getSignatureMismatches()`:

https://github.com/sjbarag/brs/blob/6613a3ee34014c38160560af70f9c822a61420b6/src/brsTypes/Callable.ts#L261-L283

Given that these are separate use cases and _may_ have different rules (we'll need to check all combinations against RBI), it's probably fine to duplicate this logic.  If we find that it's exactly the same, let's open a github issue to implement something like:

```typescript
function tryCoerceType(into: ValueKind, value: BrsType): BrsType {
}
```

that covers all combinations.